### DISCUSSION TITLE
Edges with code labels

### DISCUSSION THREAD

**Discussion by canelhasmateus** [terrastruct/d2#1561] [Q&A]
Is there any way to get `syntax highlighting` in edge labels? I would expect the following to work, but it doesn't quite so.
It would be pretty useful to indicate schemas and payloads, specially in sequence diagrams. 

https://play.d2lang.com/?script=hM9BaoYwEAXgfU7xyPrvBUJxIdqV0FJ10VUJOkhKomCGojTevSRUS6nyZxGY94aBb8FDhlXhSwCJoBDCh59GASCl8UnTSyjUPJtxuGFPPTFbcjTye6-ZpEI1ddoWmum277ip19bwKhXqsqpC3r4dHdPsYv7y3ISntmlfy99KL-TjQeP5sYlDdnSfxFIhN0NBnXHa_uSbAEIQm_jL-e8509y3XEsuHdeKc8OW_mT4DgAA__8%3D&

**Comment by alixander** [comment]
@canelhasmateus wouldn't that just be the same as another code node though?

https://play.d2lang.com/?script=hJHBasMwEETv-opB5_QHRMkhJD0ZWhr70FNZoiWoSFaJlhJR-d-LjO1ebHIRmpllmcfe8bTHN2UfydZvVkpN0qCUrxR7BQC_4wtoZzUMznJz_XWH2U0s4jlwL5-WhLVBEy_kjyS8m2dCtOSdZG1wPjVNOXQfSyZ8C9V_e23LS9d276f_iO6c6kKX5LmtYr9kPyza4OCuR764QH7yBwWUorIZa4-XWYFZQ3kMso2xCbGNsA4w1R_UXwAAAP__&

![Screen Shot 2023-08-21 at 11 26 08 AM](https://github.com/terrastruct/d2/assets/3120367/1a61dc56-1c86-441e-982d-af86131dfab7)

**Reply by alixander** [reply]
You can further make it more look like a pass-through by dropping that first arrowhead:

![Screen Shot 2023-08-21 at 11 26 45 AM](https://github.com/terrastruct/d2/assets/3120367/277aa657-0b90-4c02-9f03-3246ff997351)

**Comment by canelhasmateus** [comment]
This is usually the workaround I use for nodes.

However, specifically for `sequence_diagrams`, there is no possibility of doing so.
The first commented block tries to do as you say, which creates a new participant ( with code highlighting ) 
The second commented block is one of the workarounds I tried, which creates a note, which also don't have highlighting. 

My current approach has been to specify `style.font: mono`, which lacks the highlighting, but is better than the default for my use cases ( detail request/response schemas ) 

https://play.d2lang.com/?script=vJA9rsMgEIT7PcUI1_YBKN5VntYB5UfYJJAosgJ3j8CL3bgODaNv0H4rpqWP9vGy88lqfAioK2k0-G-ufA48EcDo_zAOoTTxqSGBsMPygtc5QP0RjXSLfq4AUpSjLtY5rzTU2wdnlBS53olKIqADMJahd16cZ1PnCxe02jocGDvs1pYPza3MW06Sti14-JEv0zcAAP__&

**Reply by alixander** [reply]
ohh good point, that's a common enough use case. can you file this as an issue please?
